src/layers/base-layer.js

```diff
@@ -478,14 +482,18 @@ export default class BaseLayer {
   // MAP LAYER FUNCTIONALITY
   setViewport() {
     const {
-      width, height, latitude, longitude, zoom,
+      width, height, latitude, longitude, zoom, pitch, bearing, altitude,
```

Comment:
> use pitch=0, bearing=0, altitude=1.5 for backwards compatibility?

Good point, I just don't want to spread out such constants in multiple places (especially the default altitude, which is matched to mapbox's default settings).

I moved the defaults to the Viewport class and made it able to apply default to 'undefined' params (as opposed to just omitted params).